com.unity.netcode.gameobjects/Runtime/Serialization/BytePacker.cs

```diff
@@ -0,0 +1,617 @@
+using System;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// Utility class for packing values in serialization.
+    /// </summary>
+    public static class BytePacker
+    {
+        #region Managed TypePacking
+
+        /// <summary>
+        /// Writes a boxed object in a packed format
+        /// Named differently from other WriteValuePacked methods to avoid accidental boxing.
+        /// Don't use this method unless you have no other choice.
+        /// </summary>
+        /// <param name="writer">Writer to write to</param>
+        /// <param name="value">The object to write</param>
+        /// <param name="isNullable">
+        /// If true, an extra byte will be written to indicate whether or not the value is null.
+        /// Some types will always write this.
+        /// </param>
+        public static void WriteObjectPacked(ref FastBufferWriter writer, object value, bool isNullable = false)
```

Comment:
since we're not offering anything automagical under this method here (apart from auto type -> method convenience), would it make more sense to drop this `WriteObject`/`ReadObject` combo entirely? I think we could have a convo about this and hopefully I might convince people to do so :)

Reply:
See above comment. I'm hoping the answer is yes but don't want to commit to that until I've converted the existing messages without needing it.